src/main/java/org/takes/rq/RqLive.java

```diff
@@ -68,53 +63,30 @@ public RqLive(final InputStream input) throws IOException {
      * @throws IOException If fails
      * @checkstyle ExecutableStatementCountCheck (100 lines)
      */
-    @SuppressWarnings
-        (
-            {
-                "PMD.AvoidInstantiatingObjectsInLoops",
-                "PMD.StdCyclomaticComplexity",
-                "PMD.ModifiedCyclomaticComplexity"
-            }
-        )
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     private static Request parse(final InputStream input) throws IOException {
         boolean eof = true;
-        final Collection<String> head = new LinkedList<String>();
+        final Collection<String> head = new LinkedList<>();
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        Opt<Integer> data = new Opt.Empty<Integer>();
-        while (true) {
-            data = RqLive.data(input, data);
-            if (data.get() < 0) {
-                break;
-            }
+        Opt<Integer> data = new Opt.Empty<>();
+        data = RqLive.data(input, data, false);
+        while (data.get() > 0) {
             eof = false;
             if (data.get() == '\r') {
-                if (input.read() != '\n') {
-                    throw new HttpException(
-                        HttpURLConnection.HTTP_BAD_REQUEST,
-                        String.format(
-                            // @checkstyle LineLength (1 line)
-                            "there is no LF after CR in header, line #%d: \"%s\"",
-                            head.size() + 1, new String(
-                                baos.toByteArray(), StandardCharsets.UTF_8
-                            )
-                        )
-                    );
-                }
+                RqLive.checkLineFeed(input, baos, head.size() + 1);
                 if (baos.size() == 0) {
                     break;
                 }
-                data = new Opt.Single<Integer>(input.read());
+                data = new Opt.Single<>(input.read());
                 final Opt<String> header = RqLive.newHeader(data, baos);
                 if (header.has()) {
                     head.add(header.get());
                 }
+                data = RqLive.data(input, data, false);
```

Comment:
@mkordas same answer as above, if you check well I strictly respected the previous method flow because if you don't you will get failing tests.